### PR TITLE
Deploy Lambda via GitHub Actions

### PR DESCRIPTION
**Pull request by samestep**
I created the AWS access key via my IAM user.

### CODE REVIEW

**Comment by seemethere** [comment]
Should we put this behind a github environment?

**Comment by samestep** [comment]
> Should we put this behind a github environment?

@seemethere Sure, that sounds fine if you know how to do it.

**Comment by seemethere** [comment]
> > Should we put this behind a github environment?
> 
> @seemethere Sure, that sounds fine if you know how to do it.

Oh whoops, looks like someone might need admin access in order to do something like that

**Review by seemethere — APPROVED** [review]

File: .github/workflows/deploy_lambda.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Deploy Lambda
+
+on:
+  push:
+    branches:
+      - master
```
Comment (seemethere) [inline .github/workflows/deploy_lambda.yml:6]: Actually can we set this only to deploy if there are changes to the lambda itself?

Comment (samestep) [reply .github/workflows/deploy_lambda.yml:6]: Makes sense, I'll do that

Comment (seemethere) [reply .github/workflows/deploy_lambda.yml:6]: Similar to:

https://github.com/pytorch/builder/blob/da1928b8e919c362ada44ef5a9b2e10bc7f518f8/.github/workflows/build-conda-images.yml#L3-L7

```
on:
  pull_request:
    paths:
      - conda/Dockerfile
      - 'common/*'
 ```

**Review by seemethere — APPROVED** [review]

**Pull request merged**